lib/database_handler.rb

```diff
@@ -0,0 +1,26 @@
+require 'pg'
+
+class Database
+  def self.connect 
+    return PG.connect(dbname: 'chitter_test') if ENV['RACK_ENV'] == 'test'  
+    return PG.connect(dbname: 'chitter')
+  end 
+
+  def self.all
+    connection = Database.connect
+    result = connection.exec('SELECT * FROM peeps ORDER BY id DESC;')
+  end 
+
+  def self.all_filtered(filter:)
+    connection = Database.connect
+    result = connection.exec("SELECT * FROM peeps WHERE messages LIKE '%#{filter}%'")
+  end 
+
+  def self.add(message:)
+    connection = Database.connect
+    time = Time.now
+    time = time.strftime("%d/%m/%Y")
```

Comment:
Storing time as a string works but as you can see, it requires you to explicitly format it before putting it in the database. That ties your View to your Model -- ideally you would store all values in a format that isn't tied to how you want to present it to the user so that you can easily change the latter if needed without changing your database tables. For example, what if later there was a requirement that dates shown to the user should also include time information, not just the date? Your database would have no timing information for old messages.

Luckily, most databases actually provide a datatype to store time information: https://www.postgresql.org/docs/9.1/datatype-datetime.html